docs/api.md

```diff
@@ -729,19 +728,26 @@ Shortcut for [page.mainFrame().evaluate(pageFunction, ...args)](#frameevaluatepa
 #### page.evaluateHandle(pageFunction, ...args)
 - `pageFunction` <[function]|[string]> Function to be evaluated in the page context
 - `...args` <...[Serializable]|[JSHandle]> Arguments to pass to `pageFunction`
-- returns: <[Promise]<[JSHandle]>> Resolves to the return value of `pageFunction`
+- returns: <[Promise]<[JSHandle]>> Promise which resolves to the return value of `pageFunction` as in-page object (JSHandle)
 
-If the function, passed to the `page.evaluateHandle`, returns a [Promise], then `page.evaluateHandle` would wait for the promise to resolve and return its value.
+If the function passed to the `page.evaluateHandle` returns a [Promise], then `page.evaluateHandle` would wait for the promise to resolve and return its value.
 
+The only difference between `page.evaluate` and `page.evaluateHandle` is that `page.evaluate` returns a JSON representation of the object and `page.evaluateHandle` returns in-page object (JSHandle):
 ```js
-const aWindowHandle = await page.evaluateHandle(() => Promise.resolve(window));
-aWindowHandle; // Handle for the window object.
-```
+// returns a JSON object
+const jsObject = await page.evaluate(pageFunction, elementHandle);
 
-A string can also be passed in instead of a function.
+// returns in-page object (JSHandle)
+const jsHandle = await page.evaluateHandle(pageFunction, elementHandle);
+
+// both lines below print true
+console.log(`jsObject is NOT equal to jsHandle: ${jsObject !== jsHandle}`);
+console.log(`jsObject is equal to jsHandle.jsonValue(): ${jsObject === (await jsHandle.jsonValue())}`);
```

Comment:
I don't think this is actually the case: triple-equals checks for identity equality, which is not the case.

Reply:
But identity equality (aka strict equality) means that type and values are the same. In this case it shows that jsObject is the same type and value as jsHandle.jsonValue().

Reply:
@yelisaveta in case of non-primitives, it means that two objects are the same by identity, not by value.

e.g.:
```js
console.log({} === {}); // false
let a = {};
let b = a;
let c = a;
console.log(b === c); // true
```

Reply:
Sorry for a delay - I've now understood why I used strict equality check - initially I've created an example to demonstrate the difference between these two methods, see here: https://github.com/GoogleChrome/puppeteer/pull/1867/commits/53ae96405076cd21c7b7183646237a9436e3e167

Examples were using `pageFunction` which returned `element.innterHTML` - which is a String. That's why strict equality worked correctly for me. 

I see that having these snippets in the docs without the `pageFunction` definition looks confusing.